src/map/zone.cpp

```diff
@@ -1196,6 +1196,8 @@ void CZone::CharZoneOut(CCharEntity* PChar)
         PChar->StatusEffectContainer->DelStatusEffectSilent(EFFECT_LEVEL_RESTRICTION);
     }
 
+    PChar->StatusEffectContainer->DelStatusEffectsByFlag(EFFECTFLAG_ON_ZONE);
```

Comment:
I am concerned about what fragile code this call might inadvertently break, maybe better to just remove the potential SJ restriction status effect here rather than all EFFECTFLAG_ON_ZONE?

Reply:
Im not too scared - the following query shows the on_zone effect flag
`SELECT * FROM status_effects WHERE status_effects.flags & 256`

Reply:
> Im not too scared - the following query shows the on_zone effect flag `SELECT * FROM status_effects WHERE status_effects.flags & 256`

Ok, if you sure :)